src/AutoBuild.jl

```diff
@@ -1401,8 +1383,9 @@ function build_project_dict(name, version, dependencies::Array{Dependency}, juli
     for dep in dependencies
         depname = getname(dep)
         project["deps"][depname] = string(jll_uuid(depname))
-        if has_compat_info(dep)
-            project["compat"][depname] = string(exactly_this_version(dep.pkg.version))
+        if dep isa Dependency && length(dep.compat) > 0
+            Pkg.Types.semver_spec(dep.compat) # verify dep.compat is valid
```

Comment:
Isn't this enforced in [the constructor of `Dependency`](https://github.com/fingolfin/BinaryBuilderBase.jl/blob/e2f1baefa710b5d3e44e61fa30a1259e34fd3f58/src/Dependencies.jl#L77)?  I guess double checking isn't a big deal though

Reply:
Thing is, this data is produced in another package, then serialized, deserialized, then processed here, possibly even on another machine / VM. So I prefer to code defensively.

Reply:
Fair enough